packages/cli/api-importers/openapi/openapi-ir-to-fern/src/OpenApiIrConverterContext.ts

```diff
@@ -49,22 +62,54 @@ export class OpenApiIrConverterContext {
     public detectGlobalHeaders: boolean;
     public objectQueryParameters: boolean;
     public respectReadonlySchemas: boolean;
+    public onlyIncludeReferencedSchemas: boolean;
+
     private enableUniqueErrorsPerEndpoint: boolean;
     private defaultServerName: string | undefined = undefined;
     private unknownSchema: Set<number> = new Set();
+
+    /**
+     * The set of referenced schema ids to include in the generated definition.
+     * If this value is undefined, _all_ schemaIds should be treated as referenced,
+     * and therefore included in the generated definition.
+     */
+    private referencedSchemaIds: Set<SchemaId> | undefined;
+
     /**
      * The current endpoint method being processed. This is used to determine
      * whether certain properties should be included in the generated definition
      * (e.g. readonly properties are excluded for POST/PUT endpoints).
      */
     private endpointMethod: HttpMethod | undefined;
+
+    /**
+     * Whether the current schema being processed is part of an endpoint.
+     * This is used to determine whether certain properties should be included
+     * in the generated definition (e.g. endpoint tree-shaking).
+     */
+    private inEndpoint = false;
+
     /**
      * Whether the current schema being processed is part of a request body.
      * This is used to determine whether certain properties should be included
      * in the generated definition (e.g. readonly properties are excluded for request bodies).
      */
     private inRequest = false;
 
+    /**
+     * Whether the current schema being processed is part of a webhook.
+     * This is used to determine whether certain properties should be included
+     * in the generated definition (e.g. webhook tree-shaking).
+     */
+    private inWebhook = false;
+
+    /**
+     * Whether the current schema being processed is part of a websocket channel.
+     * This is used to determine whether certain properties should be included
+     * in the generated definition (e.g. channel tree-shaking).
+     */
+    private inChannel = false;
```

Comment:
I kept things in their current state for consistency, but we can also roll out an enum-based approach if we want to reduce some of the copying between the `isIn{Channel,Endpoint,Webhook}` helper functions.

I imagine we'd want something like:

```typescript
enum State {
  Endpoint,
  Channel,
  Webhook,
  Request,
}
```

```typescript
private inState: State | undefined;
```

```typescript
/**
 * Sets that we're currently processing an endpoint
 */
public setInEndpoint(): void {
    this.setInState(State.Endpoint);
}

private setInState(state: State): void {
    this.inState = state;
}
```

Reply:
Decided to go with this approach after all!